test/integration/certificates/generates_new_private_key_per_request_test.go

```diff
@@ -171,12 +187,9 @@ func TestGeneratesNewPrivateKeyIfMarkedInvalidRequest(t *testing.T) {
 		t.Fatalf("failed to parse first CSR: %v", err)
 	}
 
-	pk1 := csr1.PublicKey.(crypto.PublicKey)
-	pk2 := csr2.PublicKey.(crypto.PublicKey)
-
-	if pk1.(comparablePublicKey).Equal(pk2) {
-		t.Errorf("expected the two requests to have been signed by distinct private keys, but the private key has been reused")
-	}
+	match, err := pki.PublicKeysEqual(csr1.PublicKey, csr2.PublicKey)
+	require.NoError(t, err)
+	assert.False(t, match, "expected the two requests to have been signed by distinct private keys, but the private key has been reused")
```

Comment:
I found this existing helper function (`pki.PublicKeysEqual`)  in use in the E2E tests, so I decided to make use of it here.